ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -175,69 +246,64 @@ public LogicalSchema withAlias(final String alias) {
    * @return the schema without any aliases in the field name.
    */
   public LogicalSchema withoutAlias() {
-    final SchemaBuilder newSchema = SchemaBuilder
-        .struct()
-        .name(schema.name());
-
-    for (final Field field : schema.fields()) {
-      final String unaliased = SchemaUtil.getFieldNameWithNoAlias(field.name());
-      newSchema.field(unaliased, field.schema());
+    if (!alias.isPresent()) {
+      throw new IllegalStateException("Not aliased");
```

Comment:
why not return `this`?

Reply:
Because this should never happen in the way KSQL works.   If that changes we can change the function. But for now the type is putting sanity checks in place to ensure people don't change code and inadvertently do the wrong thing.